lxd/db/images.go

```diff
@@ -1138,6 +1138,17 @@ WHERE images.fingerprint = ?
 	return c.getNodesByImageFingerprint(q, fingerprint)
 }
 
+// GetNodesWithImageAndAutoUpdate returns the addresses of online nodes which already have the image.
+func (c *Cluster) GetNodesWithImageAndAutoUpdate(fingerprint string, autoUpdate bool) ([]string, error) {
+	q := `
+SELECT DISTINCT nodes.address FROM nodes
+  LEFT JOIN images_nodes ON images_nodes.node_id = nodes.id
```

Comment:
@monstermunchkin from what I can see we aren't interested in the nodes that dont have this image, so is there a reason for the use of LEFT JOIN rather than just a straight JOIN here?

Reply:
@tomponline I just copied it from the function above :)

Reply:
@monstermunchkin does it work as expected using just `JOIN` as I've tended to avoid using `LEFT JOIN` unless I specifically wanted the behaviour it brings (i.e return all left hand side rows and return null for the columns on the right hand side if they dont match the join condition), this way it makes it clear that we're not interested in the rows if they don't match, and for larger result sets there is likely to be a performance gain by not requiring sqlite to generate all the rows in the result set. I realise in this case the query's WHERE statement is filtering out the null rows by filtering on the `images` table, but it feels a bit counter-intuitive to specify a LEFT JOIN (indicating we want all rows) and then filtering them out :).

If it were an existing query I'd probably leave it alone unless specifically working on it, but as this is a new query if it works as desired with a normal JOIN then its probably worth using it.

Reply:
It works as expected with just `JOIN`. I'll update the commit.